Src/FluentAssertions/Common/NonGenericCollectionWrapper.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace FluentAssertions.Common
+{
+    internal static class NonGenericCollectionWrapper
```

Comment:
🔧 (suggestion) I think it'll make the code even easier to understand if we name this `ReadOnlyNonGenericCollectionWrapper`